From 36357b79af717705b61353f554b3526fb0d0f070 Mon Sep 17 00:00:00 2001
From: benrichard-amd <ben.richard@amd.com>
Date: Fri, 7 Aug 2026 21:28:34 -0400
Subject: [PATCH 2/2] [PowerPC] Add GCC toolset 14/15 detection

Cherry-picked from ROCm/llvm-project#3810.

Adds gcc-toolset-14 and gcc-toolset-15 to Clang driver RHEL/CentOS
GCC toolset search prefixes.
---
diff --git a/clang/lib/Driver/ToolChains/Gnu.cpp b/clang/lib/Driver/ToolChains/Gnu.cpp
index ab997eb5a7eb..54bc443fc3d1 100644
--- a/clang/lib/Driver/ToolChains/Gnu.cpp
+++ b/clang/lib/Driver/ToolChains/Gnu.cpp
@@ -2308,6 +2308,8 @@ void Generic_GCC::GCCInstallationDetector::AddDefaultGCCPrefixes(
       D.getVFS().exists("/opt/rh")) {
     // TODO: We may want to remove this, since the functionality
     //   can be achieved using config files.
+    Prefixes.push_back("/opt/rh/gcc-toolset-15/root/usr");
+    Prefixes.push_back("/opt/rh/gcc-toolset-14/root/usr");
     Prefixes.push_back("/opt/rh/gcc-toolset-13/root/usr");
     Prefixes.push_back("/opt/rh/gcc-toolset-12/root/usr");
     Prefixes.push_back("/opt/rh/gcc-toolset-11/root/usr");
-- 
2.52.0
